homeassistant/components/lamarzocco/__init__.py

```diff
@@ -39,10 +134,51 @@ async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
     if unload_ok:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate config entry."""
+    if entry.version > 2:
+        # guard against downgrade from a future version
+        return False
+
+    if entry.version == 1:
+        cloud_client = LaMarzoccoCloudClient(
+            username=entry.data[CONF_USERNAME],
+            password=entry.data[CONF_PASSWORD],
+        )
+        try:
+            fleet = await cloud_client.get_customer_fleet()
+        except (AuthFail, RequestNotSuccessful) as exc:
+            _LOGGER.error("Migration failed with error %s", exc)
+            return False
+
+        assert entry.unique_id is not None
+        device = fleet[entry.unique_id]
+        v2_data = {
+            CONF_USERNAME: entry.data[CONF_USERNAME],
+            CONF_PASSWORD: entry.data[CONF_PASSWORD],
+            CONF_MODEL: device.model,
+            CONF_NAME: device.name,
+            CONF_TOKEN: device.communication_key,
+        }
+
+        if CONF_HOST in entry.data:
+            v2_data[CONF_HOST] = entry.data[CONF_HOST]
+
+        if CONF_MAC in entry.data:
+            v2_data[CONF_MAC] = entry.data[CONF_MAC]
+
+        entry.version = 2
```

Comment:
This is not correct, the new version should be passed to `async_update_entry`

Reply:
done. But out of curiousity, what difference does it make?